src/Features/CSharp/Portable/TypeStyle/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -51,6 +53,13 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             SyntaxNode node, CancellationToken cancellationToken)
         {
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (node is VariableDeclarationSyntax variableDeclaration)
+            {
+                // CS0822 is reported on the declaration, not the type.
+                node = variableDeclaration.Type;
+            }
```

Comment:
Do not like this.  We're effectively munging together different diagnostics, then trying to piece together what is going on on the other end.